provider/resources.go

```diff
@@ -5459,6 +5459,136 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
+								Fields: map[string]*tfbridge.SchemaInfo{
+									"apply_server_side_encryption_by_default": {
+										Name:        "applyServerSideEncryptionByDefaults",
+										MaxItemsOne: tfbridge.False(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"lifecycle_rule": {
+				Elem: &tfbridge.SchemaInfo{
+					Fields: map[string]*tfbridge.SchemaInfo{
+						"expiration": {
+							MaxItemsOne: tfbridge.False(),
+							Name:        "expirations",
+						},
+						"noncurrent_version_expiration": {
+							MaxItemsOne: tfbridge.False(),
+							Name:        "noncurrentVersionExpirations",
+						},
+					},
+				},
+			},
+			"object_lock_configuration": {
+				Elem: &tfbridge.SchemaInfo{
+					Fields: map[string]*tfbridge.SchemaInfo{
+						"rule": {
+							Name:        "rules",
+							MaxItemsOne: tfbridge.False(),
+							Elem: &tfbridge.SchemaInfo{
+								Fields: map[string]*tfbridge.SchemaInfo{
+									"default_retention": {
+										Name:        "defaultRetentions",
+										MaxItemsOne: tfbridge.False(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"replication_configuration": {
+				MaxItemsOne: tfbridge.False(),
+				Name:        "replicationConfigurations",
+				Elem: &tfbridge.SchemaInfo{
+					Fields: map[string]*tfbridge.SchemaInfo{
+						"rules": {
+							Elem: &tfbridge.SchemaInfo{
+								Fields: map[string]*tfbridge.SchemaInfo{
+									"filter": {
+										MaxItemsOne: tfbridge.False(),
+										Name:        "filters",
+									},
+									"source_selection_criteria": {
+										MaxItemsOne: tfbridge.False(),
+										Name:        "sourceSelectionCriterias",
+										Elem: &tfbridge.SchemaInfo{
+											Fields: map[string]*tfbridge.SchemaInfo{
+												"sse_kms_encrypted_objects": {
+													Name:        "sseKmsEncryptedObjects",
+													MaxItemsOne: tfbridge.False(),
+												},
+											},
+										},
+									},
+									"destination": {
+										MaxItemsOne: tfbridge.False(),
+										Name:        "destinations",
+										Elem: &tfbridge.SchemaInfo{
+											Fields: map[string]*tfbridge.SchemaInfo{
+												"metrics": {
+													MaxItemsOne: tfbridge.False(),
+												},
+												"replication_time": {
+													MaxItemsOne: tfbridge.False(),
+													Name:        "replicationTimes",
+												},
+												"access_control_translation": {
+													MaxItemsOne: tfbridge.False(),
+													Name:        "accessControlTranslations",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Aliases: []tfbridge.AliasInfo{
+			{
+				Type: ref("aws:s3/bucket:Bucket"),
+			},
+		},
+	}
+	prov.Resources["aws_s3_bucket_legacy"] = legacyInfo
+	err := shim.CloneResource(prov.P.ResourcesMap(), "aws_s3_bucket", "aws_s3_bucket_legacy")
+	contract.AssertNoErrorf(err, "Failed to rename the resource")
+
```

Comment:
Just for my own edification: I couldn't follow what part of the changes here reintroduces the various BucketV2 adjacent resources. (eg. BucketAclV2)

Reply:
It's the changes to the `bridge-metadata.json`. Changing the version to `v7` means that they are kept since it thinks the change is occurring in a minor version.